### PR TITLE
TTOOLS-709 Addition of Preview component

### PR DESCRIPTION
**Pull request by mdrillin**
Additions to the expandable preview panel
- Created PreviewResults component and story.  This component contains just the query results and is used in the ExpandablePreview component
- Updated the ExpandablePreview component to include toggle buttons, refresh button and added necessary i18ns
- queryRow and queryColumn functions were moved to VirtualizationUtils so they could be reused in multiple components.
- adapt ViewEditorOutputPage and ViewEditorSqlPage to utilize the updated ExpandablePreview component.
- resolvers updated so that previewResults and button selection can be maintained between page changes
- updated ViewEditorNavBar to include a 'Done' button and adapt to resolvers additions
- simplification of getPreviewSql in VirtualizationUtils - adapts to simplifications of preview in komodo server.

### CODE REVIEW

File: app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/PreviewResults.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+   * [ { fName: 'Jean', lName: 'Frissilla', country: 'Italy' },
+   *   { fName: 'John', lName: 'Johnson', country: 'US' },
+   *   { fName: 'Juan', lName: 'Bautista', country: 'Brazil' },
+   *   { fName: 'Jordan', lName: 'Dristol', country: 'Ontario' }
+   * ]
+   */
+  queryResultRows: Array<{}>;
+  i18nEmptyResultsTitle: string;
+  i18nEmptyResultsMsg: string;
+}
```
Comment (elvisisking) [inline app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/PreviewResults.tsx]: Should there be a `loading` property maybe for when the results are being refreshed?

File: app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/ExpandablePreview.tsx [hunk 2]
```diff
@@ -10,12 +14,54 @@ import * as React from 'react';
...
+   *   { fName: 'John', lName: 'Johnson', country: 'US' },
+   *   { fName: 'Juan', lName: 'Bautista', country: 'Brazil' },
+   *   { fName: 'Jordan', lName: 'Dristol', country: 'Ontario' }
+   * ]
+   */
+  queryResultRows: Array<{}>;
+  viewDdl?: string;
+}
+
+interface IColumn {
```
[the remark this answers is not in the document]
Comment (mdrillin) [reply app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/ExpandablePreview.tsx]: Yes, good point.  I'll make that change in a follow up PR.

File: app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/PreviewResults.tsx (hunk 1 above)
Comment (mdrillin) [reply app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/PreviewResults.tsx]: Yes, that would be a good improvement.  I'll create an issue to add the loading state

File: app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/ExpandablePreview.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (elvisisking) [reply app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/ExpandablePreview.tsx]: Sounds good

File: app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/PreviewResults.tsx (hunk 1 above)
Comment (elvisisking) [reply app/ui-react/packages/ui/src/Data/Virtualizations/ViewEditor/PreviewResults.tsx]: Sounds good

**Pull request merged**